src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Parameters/SearchParameterValidator.cs

```diff
@@ -75,18 +74,44 @@ public async Task ValidateSearchParamterInput(SearchParameter searchParam, strin
             {
                 try
                 {
-                    _searchParameterDefinitionManager.GetSearchParameter(new Uri(searchParam.Url));
-
-                    // If a post, then it is a creation of a new search parameter
-                    // only allow this if no other parameters exist with the same Uri
-                    if (method.Equals(HttpPostName, StringComparison.OrdinalIgnoreCase))
+                    // If a search parameter with the same uri exists already
+                    if (_searchParameterDefinitionManager.TryGetSearchParameter(new Uri(searchParam.Url), out _))
+                    {
+                        // And if this is a request to create a new search parameter
+                        if (method.Equals(HttpPostName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            // We have a conflict
+                            validationFailures.Add(
+                                new ValidationFailure(
+                                    nameof(searchParam.Url),
+                                    string.Format(Resources.SearchParameterDefinitionDuplicatedEntry, searchParam.Url)));
+                        }
+                    }
+                    else
                     {
-                        // if no exception is thrown, then the search parameter with the same Uri was found
-                        // and this is a conflict
-                        validationFailures.Add(
-                            new ValidationFailure(
-                                nameof(searchParam.Url),
-                                string.Format(Resources.SearchParameterDefinitionDuplicatedEntry, searchParam.Url)));
+                        // Otherwise, no search parameters with a matching uri exist
+                        // Ensure this isn't a request to modify an existing parameter
+                        if (method.Equals(HttpPutName, StringComparison.OrdinalIgnoreCase) ||
+                            method.Equals(HttpDeleteName, StringComparison.OrdinalIgnoreCase))
+                        {
+                            validationFailures.Add(
+                                new ValidationFailure(
+                                    nameof(searchParam.Url),
+                                    string.Format(Resources.SearchParameterDefinitionNotFound, searchParam.Url)));
+                        }
+
+                        // Ensure the new search parameter's code value does not already exist for its base type(s)
+                        foreach (ResourceType? baseType in searchParam.Base)
+                        {
+                            if (_searchParameterDefinitionManager.TryGetSearchParameter(baseType.ToString(), searchParam.Code, out _))
```

Comment:
>baseType [](start = 88, length = 8)

Since `baseType` is nullable should we have a null check here?

Reply:
Good question. I first tried to test this by putting `null` as a base value in the search parameter json that is passed in, but it looks like it gets parsed away. 

So I then added this line right before the for-loop to see what would happen:

`searchParam.Base = new ResourceType?[] { ResourceType.Patient, null };`

and it looks like the null value gets converted to `""`. This fails gracefully since it is simply considered a code value that does not exist in the `TypeLookup` dictionary.

Let me know what you think - but I believe it should be alright to leave this as is.